week3/index.js

```diff
@@ -1,15 +1,77 @@
-const Express = require('express')
-const bodyParser = require('body-parser')
+"use strict"
+const express = require("express");
+const app = express();
+const bodyParser = require("body-parser");
+const fs = require("fs");
+const helperFunctions = require("./helperFunctions.js")
 
-const app = Express()
 
-app.use(bodyParser.json())
+app.use(bodyParser.urlencoded({ extended: false }));
 
-const {list, create, update, remove} = require('./actions')
 
-app.get('/todos', list)
-app.post('/todos', create)
-app.put('/todos/:id', update)
-app.delete('/todos/:id', remove)
+app.get("/", (req, res) => {
+    helperFunctions.showHelpMenu()
+        .then(data => {
+            req.body = data
+            res.send(req.body + "\n")
+        })
+        .catch(err => res.send(err.message))
+})
 
-app.listen(3000)
\ No newline at end of file
+app.get("/help", (req, res) => {
+    helperFunctions.showHelpMenu()
+        .then(data => {
+            req.body = data
+            res.send(req.body + "\n")
+        })
+        .catch(err => res.send(err.message))
+
+})
+
+app.get("/list", (req, res) => {
+    helperFunctions.readFileFunction("to-do-list.json")
+        .then((data) => {
+            data = data || "[]";
+            helperFunctions.myTodoList = JSON.parse(data);
+            if (helperFunctions.myTodoList.length === 0) {
+                res.send("You have no tasks to do, you are free\n");
+            } else {
+                res.send("Tasks to do: " + JSON.stringify(helperFunctions.myTodoList) + "\n");
```

Comment:
This is a bit of a problem. The entire document needs to be encoded as JSON, otherwise programs that try and decode it will fail (try doing a request in Postman to see - it won't be able to nicely format it as JSON).
In this case, it's okay to not send back a helpful message because you're dealing only in data - you're not really directly interacting with a user. The 'user' is another program that's requesting data only.
I would simply do `res.send(JSON.stringify( helperFunctions.myTodoList )`, but make sure to set the `Content-Type` header so that the 'client' program knows what type of data it's getting back.